scrapi/base/helpers.py

```diff
@@ -114,16 +114,29 @@ def oai_process_uris(*args):
         elif arg:
             identifiers.append(arg)
 
+    return identifiers
+
+
+def gather_object_uris(identifiers):
     object_uris = []
-    provider_uris = []
     for item in identifiers:
         if 'doi' in item.lower():
-            doi = item.replace('doi:', '').replace('DOI:', '').strip()
-            if 'http://dx.doi.org/' in doi:
-                object_uris.append(doi)
-            else:
-                object_uris.append('http://dx.doi.org/{}'.format(doi))
-
+            try:
```

Comment:
Also this nested exception pattern isn't great, is there any way you can make it conditional instead of exception driven?
```python
url_doi, just_doi = URL_REGEX.search(item).group(), DOI_REGEX.search(item).group()
object_uris.append(url_doi or 'http://dx.doi.org/{}'.format(just_doi.replace('doi:', '').replace('DOI:', '').strip())
```
You could also break the `'http://dx.doi.org/{}'.format(just_doi.replace('doi:', '').replace('DOI:', '').strip()` part into a nice helper function for creating DOI urls from dois, could be useful in other applications as well.